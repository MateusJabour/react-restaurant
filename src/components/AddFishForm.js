import React from 'react'

class AddFishForm extends React.Component {
  nameRef = React.createRef();
  priceRef = React.createRef();
  statusRef = React.createRef();
  descRef = React.createRef();
  imageRef = React.createRef();

  createFish = (e) => {
    e.preventDefault();
    console.log(this.nameRef);
    const fish = {
      name: this.nameRef.value.value,
      price: parseFloat(this.priceRef.value.value),
      status: this.statusRef.value.value,
      desc: this.descRef.value.value,
      image: this.imageRef.value.value,
    }
    this.props.addFish(fish);
    e.currentTarget.reset();
  }

  render () {
    return (
      <form action="" className="fish-form" onSubmit={this.createFish}>
        <input name="name" ref={this.nameRef} placeholder="Name" type="text"/>
        <input name="price" ref={this.priceRef} placeholder="Price" type="text"/>
        <textarea name="desc" ref={this.descRef} placeholder="Desc"></textarea>
        <input name="image" ref={this.imageRef} placeholder="Image" type="text"/>
        <select name="status" ref={this.statusRef}>
          <option value="available">Fresh</option>
          <option value="unavailable">Sold out</option>
        </select>
        <button type="submit">+ Add Fish</button>
      </form>
    );
  }
}

export default AddFishForm;
