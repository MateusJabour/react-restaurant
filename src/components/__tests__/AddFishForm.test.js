import React from 'react';
import { shallow } from 'enzyme';
import AddFishForm from '../AddFishForm';

it('match snapshot', () => {
  const wrapper = shallow(<AddFishForm addFish={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
  expect(1).toEqual(1);
});
