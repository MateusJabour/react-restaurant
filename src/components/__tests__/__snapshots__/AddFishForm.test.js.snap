// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<form
  action=""
  className="fish-edit"
  onSubmit={[Function]}
>
  <input
    name="name"
    placeholder="Name"
    type="text"
  />
  <input
    name="price"
    placeholder="Price"
    type="text"
  />
  <select
    name="status"
  >
    <option
      value="available"
    >
      Fresh
    </option>
    <option
      value="unavailable"
    >
      Sold out
    </option>
  </select>
  <textarea
    name="desc"
    placeholder="Desc"
  />
  <input
    name="image"
    placeholder="Image"
    type="text"
  />
  <button
    type="submit"
  >
    + Add Fish
  </button>
</form>
`;
